spec/signalfx-transform-helper-spec.js

```diff
@@ -57,25 +55,27 @@ describe('signalfx-transform-helper', () => {
     };
     const flattened = signalfxHelper.toKeyValueMap(obj);
     expect(flattened['resources_0']).to.be.equal('a');
-    expect(flattened['resources_1_b']).to.be.equal('B');
-    expect(flattened['resources_2_c_c_c']).to.be.equal('C');
+    expect(flattened['resources_1']).to.be.equal('{"b":"B"}');
+    expect(flattened['resources_2']).to.be.equal('{"c":{"c":{"c":"C"}}}');
```

Comment:
I think we still need to limit the number of properties in the flattened object. I think resources should be a single field in the flattened object.

Reply:
As I understand the number of fields is not a problem, the length of a field name is.

Reply:
So the worst case is something like "resources_999999999999999999999", which still easily fits in our 128-char field name limit.

Reply:
now **resources** are simply stringified, and in **details** objects are stringified flat values remain as is